### PR TITLE
integrate change request settings be

### PR DESCRIPTION
**Pull request by andreas-unleash**
Integrates frontend with the change request settings backedn
## About the changes

Closes #

### CODE REVIEW

File: frontend/src/component/changeRequest/changeRequest.types.ts [hunk 1]
```diff
@@ -12,6 +12,14 @@ export interface IChangeRequest {
     approvals: IChangeRequestApproval[];
 }
 
+export interface IChangeRequestEnviornmentConfig {
```
Comment (FredrikOseberg) [inline frontend/src/component/changeRequest/changeRequest.types.ts:15]: Small typo

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
-    const onClick = (enableEnvironment: string) => () => {
-        setDialogState({ isOpen: true, enableEnvironment });
+    const onClick = (enableEnvironment: string, isEnabled: boolean) => () => {
+        setDialogState({ isOpen: true, enableEnvironment, isEnabled });
     };
 
+    const onConfirm = async () => {
+        if (dialogState.enableEnvironment) {
+            await updateChangeRequestEnvironmentConfig(projectId, dialogState.enableEnvironment, !dialogState.isEnabled);
+            await refetchChangeRequestConfig()
```
Comment (FredrikOseberg) [inline frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:38]: We don't need to await here, it's ok to fetch this in the background

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 3]
```diff
@@ -1,45 +1,45 @@
...
 
-    const onClick = (enableEnvironment: string) => () => {
-        setDialogState({ isOpen: true, enableEnvironment });
+    const onClick = (enableEnvironment: string, isEnabled: boolean) => () => {
+        setDialogState({ isOpen: true, enableEnvironment, isEnabled });
     };
 
+    const onConfirm = async () => {
+        if (dialogState.enableEnvironment) {
+            await updateChangeRequestEnvironmentConfig(projectId, dialogState.enableEnvironment, !dialogState.isEnabled);
```
Comment (FredrikOseberg) [inline frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:37]: This needs to be try catched, and display a toast error if we don't succeed.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Some small changes to make here.

File: frontend/src/component/changeRequest/changeRequest.types.ts (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/changeRequest/changeRequest.types.ts:15]: Thanks

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx (hunk 3 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:37]: Done

**Review by FredrikOseberg — APPROVED** [review]

**Pull request merged**